Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/common/certificate/DccData.kt

```diff
@@ -1,8 +1,35 @@
 package de.rki.coronawarnapp.covidcertificate.common.certificate
 
+import de.rki.coronawarnapp.covidcertificate.common.decoder.RawCOSEObject
+
 data class DccData<CertT : DccV1.MetaData>(
     val header: DccHeader,
     val certificate: CertT,
     val certificateJson: String,
-    val kid: String
-)
+    val kid: String,
+    val rawCOSEObject: RawCOSEObject
```

Comment:
Close, but not quite. It is extracted from the COSE object and a byte array. Please call it signature

Reply:
so either ByteArray or ByteString

Reply:
just to get that right. `RawCOSEObject` is the right value to pass ? or what you called `signature` should be extracted from it ?

Reply:
`RawCOSEObject` is a `ByteArray` as I can see

Reply:
the signature is a byte array (but not a COSE object)

-> val signature: ByteArray

Reply:
my question here : what is the value that should be passed in ?

```kotlin
DccDate(
signature = rawCodeObject
)

// or 

DccDate(
signature = rawCodeObject.slice(0,n) // for example 
)

// or some different source ?
```

Reply:
It need to be extracted in class DccCoseDecoder as described here https://github.com/corona-warn-app/cwa-app-tech-spec/blob/proposal/signature-verification-for-dcc/docs/spec/dgc-overview-client.md#verifying-the-signature-of-a-dgc-against-a-list-of-dscs

Reply:
I think we should pass the whole COSE object not only the signature, the validation procedure requires byte representation of `header`,  `signature ` `payload` and `alg` is extracted from the` header `